### PR TITLE
Add docs about dotfiles in secret volumes

### PR DESCRIPTION
**Pull request by pmorie**
Add doc about how dotfiles are treated in secret volumes.

@bgrant0607 @erictune

### CODE REVIEW

File: docs/user-guide/secrets.md [hunk 1]
```diff
@@ -473,6 +473,67 @@ one called, say, `prod-user` with the `prod-db-secret`, and one called, say,
 }
 ```
```
Comment (pmorie) [inline docs/user-guide/secrets.md]: not sure why it thinks this markdown is nobueno, looks fine when rendered.

**Comment by bgrant0607** [comment]
Lgtm, thanks much

**Comment by bgrant0607** [comment]
Tests failed. Probably need to run update-generated-docs

**Comment by pmorie** [comment]
@bgrant0607 light is green, trap is clean

**Comment by pmorie** [comment]
Adding LGTM, as @bgrant0607 commented it was such.

**Pull request merged**